Fixed scene swapping

ReleaseTheSpice opened the pull request.

ReleaseTheSpice committed:
Fixed scene swapping

Snow106 commented on Engine/src/Engine/SceneBuilder/Scene.cpp:
```diff
@@ -74,10 +74,25 @@ namespace Engine
+		{
+			m_switch = false;
+			
+			if (score > 5) {
```
braces

Snow106 commented on Engine/src/Engine/SceneBuilder/Scene.cpp:
```diff
@@ -74,10 +74,25 @@ namespace Engine
+
 		//get a view on entities with a script Component, and execute their onUpdate.
 		const auto entities = getEntities<ScriptComponent>();
+		int count = 0;
```
we can get rid of this

Snow106 commented on Engine/src/Engine/SceneBuilder/Scene.cpp:
```diff
@@ -104,11 +119,45 @@ namespace Engine
 		Serializer::tryDeserializeScene(*this, other);
+		
+		delete m_physics;
+		glm::vec3 dimensions;
```
let's make this into a method since it's called in two places

ReleaseTheSpice committed:
addressed PR feedback

ReleaseTheSpice committed:
fixed bug where player could move onto obstacles

they had the colorswap script and physics component because I added it becaus i am dum
